modules/Microsoft.Network/connections/deploy.bicep

```
@@ -132,6 +135,7 @@ resource connection 'Microsoft.Network/connections@2022-07-01' = {
     virtualNetworkGateway2: virtualNetworkGatewayConnectionType == 'Vnet2Vnet' ? virtualNetworkGateway2 : null
     localNetworkGateway2: virtualNetworkGatewayConnectionType == 'IPsec' ? localNetworkGateway2 : null
     peer: virtualNetworkGatewayConnectionType == 'ExpressRoute' ? peer : null
+    authorizationKey: virtualNetworkGatewayConnectionType == 'ExpressRoute' ? expressRouteCircuitAuthorizationKey : null
```

Is the authorization key required or optional in case of ExpressRoute? If optional, I guess we should add a condition
```suggestion
    authorizationKey: virtualNetworkGatewayConnectionType == 'ExpressRoute' && !empty(authorizationKey) ? authorizationKey : null
```